src/NovelRT/Input/InteractionService.cpp

```diff
@@ -47,8 +49,14 @@ namespace NovelRT::Input {
   void InteractionService::acceptMouseButtonClickPush(int button, int action, const Maths::GeoVector<float>& mousePosition) {
     auto keyState = static_cast<KeyState>(action);
     auto keyCode = static_cast<KeyCode>(button);
+    auto value = mousePosition.getVec4Value() * glm::scale(glm::vec3(1920.0f / _screenSize.getX(), 1080.0f / _screenSize.getY(), 0.0f));
 
-    _mousePositionsOnScreenPerButton.at(keyCode) = mousePosition.getVec4Value() * glm::scale(glm::vec3(1920.0f / _screenSize.getX(), 1080.0f / _screenSize.getY(), 0.0f));
+    if (!_mousePositionsOnScreenPerButton.count(keyCode)) {
+      _mousePositionsOnScreenPerButton.insert({ keyCode, value });
```

Comment:
Why did this have to be moved from the constructor?

Reply:
Because any key that wasn't inserted in the constructor would cause an exception when used. So if it doesn't find it, it'll insert it directly.